Fixed Android Export issue

omkarpattanaik opened the pull request:
# Fixed ✅ : Android Export issue :bug:
## Summary of issue  :bug:
In current version of ct.js v3.2.0 the Export to Android functionality is not working even after adding JDK 11 and Android SDK.

## Reason for changes or errors or bug ⚠️
- ⚠️ Current version of capacitor project which is getting created by ct.js version 3.2.0  supports JDK 17. 
- ⚠️ @capacitor/project is causing below mentioned error 
 ```js
 Configuring the Capacitor project…
Error: ENOENT: no such file or directory, stat 'C:\Users\username\ct.js\Builds\newAppBird-android\android\app\src\main\java\MainActivity.java'
```

## Solution ✅
- [x] Added latest @capacitor/cli and @capacitor/core i.e v5.5.0  
- [x] Replaced @capacitor/project with @trapezedev/project and did necassary code changes.
- [x] Also replaced JDK version requirment from JDK 11 to JDK 17 in code and i18n.
- [x] Now Android Export produce .apk file with Taget API Level 33 which is a requirement in Google Playstore for publishing .aab or .apk file.


## Test for PR
Check list Before sending the pull request, make sure that:

- [x] ✔️ `gulp build` runs without issues.
- [x] ❌ if you develop a catmod, a test project and docs are also present. Put a test project into a catmod's directory.
- [x] ❌ if it is a work-in-progress, explicitly mark it as such in the title with the `WIP: ` prefix.
- [x] ✔️ The title of the pull request has a suitable emoji from [the Comigoji guide](https://comigo.gitlab.io/comigoji/#gitmoji). The title *will* look broken on GitHub, but it streamlines further merging and makes writing a pretty changelog easier.

## Image from Android Device

### Icon
<img src="https://github.com/ct-js/ct-js/assets/61633603/dee89954-8eca-421d-9061-6d9912908831" alt="Icon"  width="180">

### Flash Screen
<img src="https://github.com/ct-js/ct-js/assets/61633603/2c92bbd7-fa27-4c74-8a67-98b289ab66d0" alt="FlashScreen"  width="180">

### Games Play
<img src="https://github.com/ct-js/ct-js/assets/61633603/4eec4046-b3a7-471b-8c31-b4c5c83af3de" alt="GamplePlay1"  width="180">
<img src="https://github.com/ct-js/ct-js/assets/61633603/3e4133ae-bbbb-469f-8eeb-2a86751a92bc" alt="GamplePlay2"  width="180">
<img src="https://github.com/ct-js/ct-js/assets/61633603/4eb29585-27b9-4ccc-abdb-1e87e61f710f" alt="GamplePlay3"  width="180">
<img src="https://github.com/ct-js/ct-js/assets/61633603/84b92d93-fb6e-4711-893d-beea1771226a" alt="GamplePlay3"  width="180">

<br>
<br>

*** Requesting a review `@CosmoMyzrailGorynych` ***

CosmoMyzrailGorynych commented:
@omkarpattanaik Could you elaborate why you had to replace @capacitor/project with @trapezedev/project?

omkarpattanaik commented:
@CosmoMyzrailGorynych Reasons for using @trapezedev/project in brief :

1. In the latest version of Capacitor project @trapezedev/project are getting used  to to automate the configuration of Capacitor project.
Reference : [Capacitor Official Docs for Automated Configuration](https://capacitorjs.com/docs/guides/automated-configuration )

2.  Below mentioned error in ct.js (full log):
Node version used `18.7.0` , `16.19.1` and `16.20.2`.
Java version `JDK 17`
 ```js

Exporting the ct.js project…
Checking for an existing Capacitor project…
No existing projects found. Will create one.
Preparing build directories…
Creating Capacitor project…
Wrote to C:\Users\username\ct.js\Builds\newAppBird-android\package.json:

{
  "name": "newAppBird-android",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}



added 97 packages, and audited 98 packages in 21s

11 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
√ Creating capacitor.config.json in C:\Users\username\ct.js\Builds\newAppBird-android in 6.36ms
[success] capacitor.config.json created!

Next steps: 
https://capacitorjs.com/docs/getting-started#where-to-go-next
Adding Android platform…
√ Adding native android project in android in 83.26ms
√ add in 83.64ms
√ Copying web assets from web to android\app\src\main\assets\public in 46.47ms
√ Creating capacitor.config.json in android\app\src\main\assets in 624.20μs
√ copy android in 71.12ms
√ Updating Android plugins in 775.40μs
√ update android in 422.94ms
√ Syncing Gradle in 215.20μs
[success] android platform added!
Follow the Developer Workflow guide to get building:
https://capacitorjs.com/docs/basics/workflow
Configuring the Capacitor project…
Error: ENOENT: no such file or directory, stat 'C:\Users\username\ct.js\Builds\newAppBird-android\android\app\src\main\java\MainActivity.java'
```


I also tried running to device by directly using the capacitor cli `capacitor run android`. Logs for the same

 ```js
C:\Users\username\ct.js\Builds\newAppBird-android>capacitor run android

√ Copying web assets from web to android\app\src\main\assets\public in 52.86ms
√ Creating capacitor.config.json in android\app\src\main\assets in 3.87ms
√ copy android in 69.70ms
√ Updating Android plugins in 995.80μs
√ update android in 115.99ms
√ Please choose a target device: » Pixel_3c (emulator) (Pixel_3c)
× Running Gradle build - failed!
[error] Starting a Gradle Daemon (subsequent builds will be faster)

        > Configure project :app
        WARNING: Using flatDir should be avoided because it doesnt support any meta-data formats.

        > Configure project :capacitor-cordova-android-plugins
        WARNING: Using flatDir should be avoided because it doesnt support any meta-data formats.

        FAILURE: Build failed with an exception.

        * What went wrong:
        A problem occurred configuring project ':capacitor-cordova-android-plugins'.
        > Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
        > Namespace not specified. Please specify a namespace in the modules build.gradle file like so:

        android {
        namespace 'com.example.namespace'
        }

        If the package attribute is specified in the source AndroidManifest.xml, it can be migrated automatically to the
        namespace value in the build.gradle file using the AGP Upgrade Assistant; please refer to
        https://developer.android.com/studio/build/agp-upgrade-assistant for more information.

        * Try:
        > Run with --stacktrace option to get the stack trace.
        > Run with --info or --debug option to get more log output.
        > Run with --scan to get full insights.

        * Get more help at https://help.gradle.org

        Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

        You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from
        your own scripts or plugins.

        See https://docs.gradle.org/8.0.2/userguide/command_line_interface.html#sec:command_line_warnings

        BUILD FAILED in 31s 
```


In above logs I found few causes for the issue but still the causes provided were already taken care like  

- Namespace not specified - It was already provided in the ct.js 
![image](https://github.com/ct-js/ct-js/assets/61633603/d9a2ce15-2ddc-45af-aeb1-8810e1a0709c)


-  Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl - Necessary permission were given to address this issue.


This error occers in my windows 11 OS. As u can see i tried debugging the error but actual cause was not found but when I replaced the ***@capacitor/project*** with ***@trapezedev/project*** following the capacitors docs the error was resolved.
